### PR TITLE
PhaseCenterPerPort feature

### LINKED ISSUE #3547: Add PhaseCenterPerPort feature
### Description of the feature

HFSS option to set the phase center per port

### Steps for implementing the feature

_No response_

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
Close #3547

### CODE REVIEW

File: pyaedt/hfss.py [hunk 1]
```diff
@@ -5834,6 +5834,66 @@ def set_impedance_multiplier(self, multiplier):
...
+        --------
+
+        Set phase center of an antenna with two ports.
+
+        >>> hfss.set_phase_center_per_port(["Global", "Global"])
+
+        """
+
+        if not self.desktop_class.is_grpc_api:  # pragma: no cover
+            self.hfss.logger.warning("Set phase center not working with COM, user must do it manually")
```
Comment (PipKat) [inline pyaedt/hfss.py:5868]:
```suggestion
            self.hfss.logger.warning("Set phase center not working with COM. Set phase center manually")
```
Avoid "user" Use "you" instead wherever possible. In this case, the "you" is implied.

File: pyaedt/hfss.py [hunk 2]
```diff
@@ -5834,6 +5834,66 @@ def set_impedance_multiplier(self, multiplier):
...
 
+    @pyaedt_function_handler()
+    def set_phase_center_per_port(self, coordinate_system=None):
+        # type: (list) -> bool
+        """Set phase center per port.
+
+        Parameters
+        ----------
+        coordinate_system : list
+            List of coordinate system per port. The default is ``None``, in which case the
```
Comment (PipKat) [inline pyaedt/hfss.py:5845]:
```suggestion
            List of the coordinate system per port. The default is ``None``, in which case the
```

File: pyaedt/application/Design.py [hunk 3]
```diff
@@ -340,7 +339,14 @@ def boundaries(self):
             else:
                 self._boundaries[boundary] = BoundaryObject(self, boundary, boundarytype=boundarytype)
 
-        return list(self._boundaries.values()) + self.design_excitations
+        excitations = self.design_excitations
+        for exc in excitations:
+            if not self._boundaries:
+                self._boundaries[exc.name] = exc
+            elif exc.name not in list(self._boundaries.keys()):
+                self._boundaries[exc.name] = exc
```
Comment (maxcapodi78) [inline pyaedt/application/Design.py:347]:
```suggestion
        for exc in excitations:
            if not self._boundaries or exc.name not in list(self._boundaries.keys()):
                self._boundaries[exc.name] = exc
```

File: pyaedt/hfss.py [hunk 4]
```diff
@@ -5834,6 +5834,66 @@ def set_impedance_multiplier(self, multiplier):
...
+        --------
+
+        Set phase center of an antenna with two ports.
+
+        >>> hfss.set_phase_center_per_port(["Global", "Global"])
+
+        """
+
+        if not self.desktop_class.is_grpc_api:  # pragma: no cover
+            self.hfss.logger.warning("Set phase center not working with COM. Set phase center manually")
```
Comment (maxcapodi78) [inline pyaedt/hfss.py:5868]:
```suggestion
            self.hfss.logger.warning("Set phase center is not supported by AEDT COM API. Set phase center manually")
```

**Review by maxcapodi78 — COMMENTED** [review]
few minor comments

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**